synthesizer/process/src/stack/finalize_types/initialize.rs

```diff
@@ -287,19 +288,44 @@ impl<N: Network> FinalizeTypes<N> {
 
     /// Ensures the given `get` command is well-formed.
     #[inline]
-    fn check_get(
-        &mut self,
-        stack: &(impl StackMatches<N> + StackProgram<N>),
-        finalize_name: &Identifier<N>,
-        get: &Get<N>,
-    ) -> Result<()> {
-        // Ensure the declared mapping in `get` is defined in the program.
-        if !stack.program().contains_mapping(get.mapping_name()) {
-            bail!("Mapping '{}' in '{}/{finalize_name}' is not defined.", get.mapping_name(), stack.program_id())
-        }
-        // Retrieve the mapping from the program.
-        // Note that the unwrap is safe, as we have already checked the mapping exists.
-        let mapping = stack.program().get_mapping(get.mapping_name()).unwrap();
+    fn check_get(&mut self, stack: &(impl StackMatches<N> + StackProgram<N>), get: &Get<N>) -> Result<()> {
+        // Retrieve the mapping.
+        let mapping = match get.mapping() {
+            MappingLocator::Locator(locator) => {
+                // Retrieve the program ID.
+                let program_id = locator.program_id();
+                // Retrieve the mapping_name.
+                let mapping_name = locator.resource();
+
+                // Ensure the locator does not reference the current program.
+                if stack.program_id() == program_id {
+                    bail!("Locator '{locator}' does not reference an external mapping.");
+                }
+                // Ensure the current program contains an import for this external program.
+                if !stack.program().imports().keys().contains(program_id) {
+                    bail!("External program '{program_id}' is not imported by '{}'.", stack.program_id());
+                }
+                // Retrieve the program.
+                let external = stack.get_external_program(program_id)?;
+                // Ensure the mapping exists in the program.
+                if !external.contains_mapping(mapping_name) {
+                    bail!("Mapping '{mapping_name}' in '{program_id}' is not defined.")
+                }
+                // Retrieve the mapping from the program.
+                // Note that the unwrap is safe, as we have already checked the mapping exists.
+                external.get_mapping(mapping_name).unwrap()
```

Comment:
Even if it is safe, what benefit is the unwrap here?

Reply:
Ditto